include/ccf/pal/snp_ioctl6.h

```diff
@@ -47,6 +49,56 @@ namespace ccf::pal::snp::ioctl6
   };
 #pragma pack(pop)
 
+#pragma pack(push, 1)
+  // Table 20
+  struct DerivedKeyGuestFieldSelect
+  {
+    uint64_t reserved : 58;
+    uint32_t tcb_version : 1;
+    uint32_t guest_svn : 1;
+    uint32_t measurement : 1;
+    uint32_t family_id : 1;
+    uint32_t image_id : 1;
+    uint32_t guest_policy : 1;
+  };
+  static_assert(sizeof(DerivedKeyGuestFieldSelect) == 8);
+
+  // Table 19
+  struct KeySelect
+  {
+    uint32_t reserved : 29;
+    uint8_t key_sel : 2;
+    uint8_t root_key_sel : 1;
+  };
+  static_assert(sizeof(KeySelect) == 4);
+
+  struct DerivedKeyReq
+  {
+    KeySelect key_select;
+    uint32_t reserved;
+    DerivedKeyGuestFieldSelect guest_field_select;
+    uint32_t vmpl = 0;
+    uint32_t guest_svn;
+    uint64_t tcb_version;
+  }; // snp_derived_key_req in (linux) include/uapi/linux/sev-guest.h
+#pragma pack(pop)
+
+// Table 21
+#pragma pack(push, 1)
+  struct DerivedKeyResp
+  {
+    uint32_t status;
+    uint8_t reserved[0x20 - 0x04];
+    uint8_t data[32];
+  }; // snp_derived_key_req in (linux) include/uapi/linux/sev-guest.h
+  static_assert(sizeof(DerivedKeyResp) < 4000);
+  struct DerivedKeyRespWrapper
+  {
+    struct DerivedKeyResp resp;
+    uint8_t padding[4000 - sizeof(struct DerivedKeyResp)];
```

Comment:
I don't think this is right? Looking at the linked structs, the _report_ response is 4k, but the key response should be 64 bytes?

https://github.com/torvalds/linux/blob/master/include/uapi/linux/sev-guest.h#L46

```suggestion
    uint8_t padding[64 - sizeof(struct DerivedKeyResp)];
```

Reply:
Had a chat with @achamayou. The extra padding was to minimise the risk of memory corruption if the kernel updated. So now both DerivedKeyResp and AttestationResp are padded with a constant 1KB, which is then checked to ensure there hasn't been an overflow.